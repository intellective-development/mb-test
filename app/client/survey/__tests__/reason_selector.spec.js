import * as React from 'react';
import ReasonSelector from '../reason_selector';

describe('ReasonSelector', () => {
  const reasons = [
    {id: 1, name: 'tasted bad'},
    {id: 2, name: 'late delivery'}
  ];

  it('renders', () => {
    expect(render(
      <ReasonSelector
        reasons={reasons}
        hidden="false"
        reasonClicked={jest.fn()}
        selectedReasons={[reasons[1]]} />
    )).toMatchSnapshot();
  });

  it('Uses callback to update state when reason is selected', () => {
    const clickReason = jest.fn();
    const component = mount(
      <ReasonSelector
        reasons={reasons}
        hidden="false"
        reasonClicked={clickReason}
        selectedReasons={[]} />
    );

    component.find('.reason').last().simulate('click');
    expect(clickReason).toHaveBeenCalled(); // note that the component hasn't changed, because it doesn't hold the selected state
  });
});
