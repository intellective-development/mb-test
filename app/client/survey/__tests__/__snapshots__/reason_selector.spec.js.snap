// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReasonSelector renders 1`] = `
<div
  class="small-12 columns react-reasons-section false"
>
  <div
    class="large-12 header"
  >
    What went wrong?
  </div>
  <div
    class="react-reasons"
  >
    <div
      class="reason"
      role="button"
      tabindex="0"
    >
      tasted bad
    </div>
    <div
      class="reason selected"
      role="button"
      tabindex="0"
    >
      late delivery
    </div>
  </div>
</div>
`;
